### PR TITLE
[BugFix] Fix bug min/max statistics query result error

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Because the statistical information type of min and max is varchar. However, in the full collection, it is necessary to compare the size of min/max according to its real type, otherwise the result will be wrong

1. Fix bug min/max statistics query result error
2. Modify some statistics config default value

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 35 (37.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|12|00.00%|[39, 40, 68, 69, 112, 113, 115, 116, 117, 118, 119, 121]|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|5|00.00%|[39, 42, 65, 66, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|4|8|50.00%|[59, 61, 62, 63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|2|3|66.67%|[63]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 35 (37.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|12|00.00%|[39, 40, 68, 69, 112, 113, 115, 116, 117, 118, 119, 121]|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|5|00.00%|[39, 42, 65, 66, 67]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|4|8|50.00%|[59, 61, 62, 63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|2|3|66.67%|[63]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|4|4|100.00%|[]|

File: be/src/exprs/agg/histogram.h [hunk 1]
```diff
@@ -21,18 +21,22 @@ struct Bucket {
     T lower;
     T upper;
     // Up to this bucket, the total value
-    int32_t count;
+    int64_t count;
```
Comment (kangkaisen) [inline be/src/exprs/agg/histogram.h:24]: why change this again?

File: be/src/exprs/agg/histogram.h [hunk 2]
```diff
@@ -120,55 +89,157 @@ class HistogramAggregationFunction final
     std::string toBucketJson(std::string lower, std::string upper, size_t count, size_t upper_repeats,
                              double sample_ratio) const {
         return fmt::format("[\"{}\",\"{}\",\"{}\",\"{}\"]", lower, upper,
-                           std::to_string((size_t)(count * sample_ratio)),
-                           std::to_string((size_t)(upper_repeats * sample_ratio)));
+                           std::to_string((int64_t)(count * sample_ratio)),
+                           std::to_string((int64_t)(upper_repeats * sample_ratio)));
     }
 
     void finalize_to_column(FunctionContext* ctx __attribute__((unused)), ConstAggDataPtr __restrict state,
```
Comment (kangkaisen) [inline be/src/exprs/agg/histogram.h:96]: Finalize method is too complex and seems bad performance.

File: be/src/exprs/agg/histogram.h [hunk 3]
```diff
@@ -120,55 +89,157 @@ class HistogramAggregationFunction final
...
+                count++;
+            } else {
+                frequency_vec.emplace_back(std::make_pair(last_value, count));
+                count = 1;
+                last_value = v;
+            }
+        }
+        frequency_vec.emplace_back(std::make_pair(last_value, count));
+        //Sort by frequency
+        std::sort(frequency_vec.begin(), frequency_vec.end(), CmpPair<T>());
```
Comment (kangkaisen) [inline be/src/exprs/agg/histogram.h:126]: How many items need to be sorted?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 201 (19.40%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|[369, 370]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|3|00.00%|[1320, 1321, 1322]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[386]|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|6|00.00%|[48, 51, 52, 53, 56, 57]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|12|00.00%|[39, 40, 68, 69, 112, 113, 115, 116, 117, 118, 119, 121]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|0|15|00.00%|[82, 83, 97, 103, 104, 107, 108, 109, 128, 137, 138, 143, 148, 149, 150]|
|:large_blue_circle:|com/starrocks/sql/ast/DropStatsStmt.java|0|7|00.00%|[12, 13, 14, 17, 22, 27, 32]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|3|00.00%|[346, 358, 359]|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|0|1|00.00%|[40]|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|0|7|00.00%|[43, 44, 46, 47, 50, 53, 54]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|15|00.00%|[838, 839, 840, 848, 849, 850, 858, 859, 860, 1482, 1483, 1490, 1491, 1498, 1499]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJob.java|0|1|00.00%|[90]|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|1|00.00%|[31]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|0|9|00.00%|[620, 621, 622, 630, 631, 632, 640, 641, 642]|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|7|00.00%|[37, 38, 39, 42, 65, 66, 67]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|2|52|03.85%|[101, 102, 112, 113, 120, 121, 122, 123, 124, 126, 127, 128, 129, 130, 131, 143, 144, 151, 152, 153, 154, 157, 158, 159, 162, 163, 170, 171, 172, 173, 174, 175, 178, 179, 181, 182, 183, 184, 185, 186, 188, 189, 190, 191, 193, 195, 196, 197, 198, 199]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|[471, 832, 833, 835, 836, 837]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|1|5|20.00%|[96, 217, 221, 222]|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|1|3|33.33%|[107, 108]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|4|8|50.00%|[59, 61, 62, 63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|2|3|66.67%|[63]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|18|22|81.82%|[191, 202, 213, 214]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|1|1|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Pull request merged**